docs/source/advanced-usage.rst

```diff
@@ -69,3 +69,56 @@ directly.
 
     for event in connection.events():
         # As with WSConnection, only without any handshake events
+
+HTTP/2
+------
+
+WebSockets over HTTP/2 have a distinct difference to HTTP/1 in that
+only a single HTTP/2 stream is dedicated to the WebSocket rather than
+the entire connection (as in HTTP/1). This requires the HTTP/2
+connection to be managed before the WebSocket connection. `wsproto`
+offers two choices in this situation, it can manage either manage the
+HTTP/2 connection or not.
+
+Managed HTTP/2 Connection
+'''''''''''''''''''''''''
+
+Allowing `wsproto` to manage the HTTP/2 connection is the easiest way
+to support WebSockets over HTTP/2, however it restricts the entire
+connection to be for the WebSocket only i.e. no multiplexing of HTTP
+requests alongside the WebSocket. To use this simply provide the
+:class:`H2Handshake <wsproto.handshake.H2Handshake>` class to the
+`WSConnection` class on initialisation, as so,
+
+.. code-block:: python
+
+    from wsproto import ConnectionType, WSConnection
+    from wsproto.handshake import H2Handshake
+
+    client = WSConnection(ConnectionType.CLIENT, handshake_class=H2Handshake)
+    server = WSConnection(ConnectionType.SERVER, handshake_class=H2Handshake)
```

Comment:
Maybe add a comment line between these two - otherwise it looks like you need both to use it for e.g., a client-side application.